pkg/awsutils/awsutils.go

```diff
@@ -1356,9 +1403,16 @@ func (cache *EC2InstanceMetadataCache) DescribeAllENIs() (DescribeAllENIsResult,
 		if interfaceType == "trunk" {
 			trunkENI = eniID
 		}
-		if interfaceType == "efa" {
+		if interfaceType == "efa" || interfaceType == "efa-only" {
 			efaENIs[eniID] = true
 		}
+		if interfaceType != "efa-only" {
+			if len(eniMetadata.IPv4Addresses) == 0 && len(eniMetadata.IPv6Addresses) == 0 {
```

Comment:
Should this condition be a `&&` or an `||` ? 

This is the only new Error we are introducing here.  Do we strictly require this ? cc: @jayanthvn (this was introduced from the review comment).


Rest of the change looks good to me with capturing EFA only instance.

Reply:
There are 2 cases, `local-ipv4` field is missing or `local-ipv4` value is empty

`local-ipv4` value is empty -> we return error in 618

`local-ipv4` field is missing (Eg: IMDS out of sync on new node or aws-node restart) -> previously we used to error out - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/awsutils/imds.go#L293-L299 but now we return the ENI (line 640) and call describe ENI. So to keep the behavior consistent this additional check i.e, if not `efa-only` and missing primary IP we error out... But I think we don't need to check `IPv6Addresses` since we have been ignoring ipv6 missing data (need to check why that was introduced though)

Reply:
Updated. We will not be returning error if we miss ipv6addresses to maintain the behavior

Reply:
Okay, this reasoning sounds good to me.